testing/ssm_mocks.go

```diff
@@ -156,6 +157,64 @@ func (m *MockSSMClient) DescribeInstanceInformation(input *ssm.DescribeInstanceI
 		NextToken: nil,
 	}
 
-	return output, err
+	return filterDescribeInstanceInformationOutput(input, output)
+}
+
+func (m *MockSSMClient) DescribeInstanceInformationPages(input *ssm.DescribeInstanceInformationInput, fn func(*ssm.DescribeInstanceInformationOutput, bool) bool) error {
+	var err error = nil
+	var continueIterating bool = true
+
+	// Grab initial case and start looping
+	output, err := m.DescribeInstanceInformation(input)
+	for err == nil && continueIterating {
+		continueIterating = fn(output, (output.NextToken == nil))
+
+		// Just keep chugging unless we are at the end of the page.
+		if output.NextToken == nil || !continueIterating {
+			break
+		} else {
+			// continue until full list is consumed
+			input.SetNextToken(*output.NextToken)
+			output, err = m.DescribeInstanceInformation(input)
+		}
+
+	}
+
+	return err
+}
+
+func filterDescribeInstanceInformationOutput(input *ssm.DescribeInstanceInformationInput, output *ssm.DescribeInstanceInformationOutput) (*ssm.DescribeInstanceInformationOutput, error) {
+	filteredOutput := []*ssm.InstanceInformation{}
+
+	for _, instance := range output.InstanceInformationList {
+		match := true
+		for _, filter := range input.Filters {
+			if !instanceMatchesFilter(*instance, *filter) {
+				match = false
+				break
+			}
+		}
+
+		if match {
+			filteredOutput = append(filteredOutput, instance)
+		}
+	}
+
+	output.InstanceInformationList = filteredOutput
+	return output, nil
+}
+
+// instanceMatchesFilter converts the instance information to its string representation
+// and then checks that that information containes atleast one value in the
+// filter. This is a hack but should be sufficient for tests.
+func instanceMatchesFilter(instance ssm.InstanceInformation, filter ssm.InstanceInformationStringFilter) bool {
```

Comment:
This is a hack, but i figured it would be sufficient for the sake of the limited needs of the mock and would be better than working with reflection to pull out field names or hard coding them. Please let me know if you feel otherwise though.